csv-bulk-signature/src/main.ts

```diff
@@ -195,35 +161,55 @@ async function zipUpFile() {
 
   const zip = new jszip();
 
-  filesHtm.forEach((file) => {
+  filesHtm.forEach(async (file) => {
     try {
       const filePath = `${folderPath}/${file}`;
-      const fileContents = fs.readFileSync(filePath);
+      const fileContents = await fs.promises.readFile(filePath);
       zip.file(file, fileContents);
     } catch (error: any) {
       console.error(`Error adding file ${file}: ${error.message}`);
     }
   });
 
   const buffer = await zip.generateAsync({ type: "nodebuffer" });
-  fs.writeFileSync(
-    `./dist/signatures-${
+  await fs.promises.writeFile(
+    `${folderPath}-${
       new Date().toISOString().split(":").join("_").split(".")[0] // date and time
     }.zip`,
     buffer
   );
 }
 
-function createStatusReport() {
+async function createStatusReport() {
   skippedRows.unshift("\nROWS/SIGNATURES SKIPPED:\n");
   processedRows.unshift("\nROWS/SIGNATURES PROCESSED SUCCESSFULLY:\n");
   const combinedStatus = skippedRows.concat(processedRows);
-  fs.writeFile(
+  await fs.promises.writeFile(
     "./dist/signatures/_statusReport.txt",
-    combinedStatus.join("\n"),
-    (err) => {
-      if (err) throw err;
-      console.log("Report created 👍");
-    }
+    combinedStatus.join("\n")
   );
+  console.log("Report created 👍");
 }
+
+async function getCSVRows(path: string) {
+  return new Promise((resolve: (arg: Contact[]) => void, reject) => {
+    return fs.promises.readFile(path).then((fileData) => {
+      parse(fileData, { columns: true }, function (err, rows) {
+        resolve(rows as Contact[]);
+      });
+    });
+  });
+}
+
+async function main() {
+  await setupFolders();
+  const template = await getTemplate();
+  await checkHeadersToBeSame();
+  await generateSignatures(template);
+  await createStatusReport(); // refactoring the results specifics to be shown in a text file, instead
+  await zipUpFile();
+}
```

Comment:
create main() as the initial function for all async steps in the generate process